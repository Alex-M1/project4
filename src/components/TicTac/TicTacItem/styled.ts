import styled from 'styled-components';

export const StTicTacItem = styled.div`
  border: 1px solid black;
  width: 33.333%;
  height: 33.333%;
  padding: 10px;
`;
