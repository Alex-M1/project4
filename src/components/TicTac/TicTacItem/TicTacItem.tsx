import React from 'react';
import { useTheme } from 'src/components/hooks/useTheme';
import { StTicTacItem } from './styled';

interface IProps {
  square: number;
  content: string;
  onClick: (square: number) => void
}

const TicTacItem: React.FC<IProps> = ({ square, content, onClick }) => {
  const handleClick = () => onClick(square);
  const { colors, theme } = useTheme();
  return (
    <StTicTacItem
      theme={theme}
      colors={colors}
      onClick={content ? null : handleClick}
    >
      {content}
    </StTicTacItem>
  );
};

export default TicTacItem;
