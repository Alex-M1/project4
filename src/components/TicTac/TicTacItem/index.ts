import TicTacItem from './TicTacItem';

export default TicTacItem;
