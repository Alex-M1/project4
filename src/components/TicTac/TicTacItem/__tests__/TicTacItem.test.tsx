import { mount, shallow } from 'enzyme';
import React from 'react';
import TicTacItem from '../TicTacItem';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);

describe('TicTacItem', () => {
  let props;
  beforeEach(() => {
    props = {
      square: 2,
      content: null,
      isGameEnd: false,
      onClick: jest.fn(),
    };
  });

  it('Should match snapshot', () => {
    const component = shallow(<TicTacItem {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StTicTacItem', () => {
    const component = mount(<TicTacItem {...props} />);
    expect(component.find('styled__StTicTacItem')).toHaveLength(1);
  });
  it('should click StTicTacItem', () => {
    const component = mount(<TicTacItem {...props} />);
    component.find('styled__StTicTacItem').simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
  it('should click StTicTacItem content != null', () => {
    props.content = 'ada';
    const component = mount(<TicTacItem {...props} />);
    component.find('styled__StTicTacItem').simulate('click');
    expect(props.onClick).not.toHaveBeenCalled();
  });
  it('should click StTicTacItem isGameEnd', () => {
    props.isGameEnd = true;
    const component = mount(<TicTacItem {...props} />);
    component.find('styled__StTicTacItem').simulate('click');
    expect(props.onClick).not.toHaveBeenCalled();
  });
});
