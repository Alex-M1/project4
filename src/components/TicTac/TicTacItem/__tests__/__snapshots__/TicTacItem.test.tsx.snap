// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacItem Should match snapshot 1`] = `"<div class=\\"styled__StTicTacItem-erJWA-d cemGHT\\"></div>"`;
